library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -68,6 +66,8 @@ void download(final Callback callback) {
         filePath = result.filePath();
         fileSize.setCurrentSize(filePersistence.getCurrentSize());
 
+        callback.onUpdate(downloadFileStatus);
```

Comment:
Performing an initial update that might be error, see removal from line 50 above. Instead we now wait until we have updated all of the status and then emit before performing the potentially long running `Sync` below.